File: firebase-api-library/src/androidTest/java/com/fazemeright/firebase_api_library/api/UserAuthenticationTest.java
package com.fazemeright.firebase_api_library.api;

import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertNull;

import androidx.test.ext.junit.runners.AndroidJUnit4;
import com.fazemeright.firebase_api_library.api.firebase.FireBaseUserAuthentication;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;

@RunWith(AndroidJUnit4.class)
public class UserAuthenticationTest {
  public UserAuthentication userAuthentication;
  public final String EMAIL_ADDRESS = "test@test.com";
  public final String PASSWORD = "password123";

  @Before
  public void setUp() {
    userAuthentication = FireBaseUserAuthentication.getInstance();
  }

  @Test
  public void signOutUser() {
    assertNotNull(userAuthentication.getCurrentUserUid());
    userAuthentication.signOutUser();
    assertNull(userAuthentication.getCurrentUserUid());
  }

  @Test
  public void isUserVerified() {
  }

  @Test
  public void createNewUserWithEmailPassword() {
  }

  @Test
  public void signInWithEmailAndPassword() {
  }

  @Test
  public void sendEmailVerification() {
  }

  @Test
  public void getCurrentUserEmail() {
  }

  @Test
  public void sendPasswordResetEmail() {
  }

  @Test
  public void reloadCurrentUserAuthState() {
  }

  @Test
  public void getCurrentUserUid() {
  }

  @Test
  public void getCurrentUser() {
  }

  @Test
  public void getUserName() {
  }
}

File: firebase-api-library/src/androidTest/java/com/fazemeright/library/api/UserAuthenticationTest.kt
package com.fazemeright.library.api

import androidx.test.ext.junit.runners.AndroidJUnit4
import com.fazemeright.library.api.domain.authentication.UserAuthentication
import com.fazemeright.library.api.domain.authentication.firebase.FireBaseUserAuthentication
import org.junit.Assert
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class UserAuthenticationTest {
    var userAuthentication: UserAuthentication? = null
    val EMAIL_ADDRESS = "test@test.com"
    val PASSWORD = "password123"
    @Before
    fun setUp() {
        userAuthentication = FireBaseUserAuthentication.instance
    }

    @Test
    fun signOutUser() {
        Assert.assertNotNull(userAuthentication!!.currentUserUid)
        userAuthentication!!.signOutUser()
        Assert.assertNull(userAuthentication!!.currentUserUid)
    }

    @get:Test
    val isUserVerified: Unit
        get() {}

    @Test
    fun createNewUserWithEmailPassword() {
    }

    @Test
    fun signInWithEmailAndPassword() {
    }

    @Test
    fun sendEmailVerification() {
    }

    @get:Test
    val currentUserEmail: Unit
        get() {}

    @Test
    fun sendPasswordResetEmail() {
    }

    @Test
    fun reloadCurrentUserAuthState() {
    }

    @get:Test
    val currentUserUid: Unit
        get() {}

    @get:Test
    val currentUser: Unit
        get() {}

    @get:Test
    val userName: Unit
        get() {}
}